proxy/coco/src/project/checkout.rs

```diff
@@ -206,15 +284,18 @@ where
                         path.join(name)
                     }
                 });
+        crate::project::ensure_directory(&project_path)?
+            .ok_or_else(|| Error::AlreadExists(project_path.clone()))?;
 
         // Clone the repository
-        let mut builder = git2::build::RepoBuilder::new();
-        builder.branch(self.default_branch.as_str());
-        let repo = ownership.clone(self.urn, &self.default_branch, &project_path, &mut builder)?;
+        let (repo, rad) =
+            ownership.clone(open_storage, self.urn, &self.default_branch, &project_path)?;
 
         // Set configurations
-        super::set_rad_upstream(&repo, &self.default_branch)?;
+        super::set_upstream(&repo, &rad, self.default_branch.clone())?;
         include::set_include_path(&repo, self.include_path)?;
+        repo.set_head(Qualified::from(self.default_branch).as_str())?;
+        repo.checkout_head(None)?;
```

Comment:
I’m not sure this plays well with existing repos that have a dirty working tree, a different branch checked out, etc.

Reply:
It shouldn't get here if it's already a project though https://github.com/radicle-dev/radicle-upstream/pull/1463/files/cd99c84c65040be8565cd0b400655f877f88bf95#diff-3362a0bfada6f68cdf7b17ea16658dbbfe028d7ce27b96bf1a4fd6e37318b8c8R288